### DISCUSSION TITLE
Reasoning behind 308 redirects when trailingSlash: true

### DISCUSSION THREAD

**Discussion by alejandronanez** [vercel/next.js#15432] [Help] [answered]
👋 I just upgraded the latest canary build for 9.4.5 and I noticed that `trailingSlash` is available 🎉.

I'd like to know what's the reasoning behind using a 308 for those redirects and not a 301, or even let the consumer decide what kind of redirect they want to use to remove/persist the trailing slash.

My guess is that this isn't configurable because of time constrains, if that's the case, maybe I can give you a hand with that? 

Thank you!

**Comment by timneutkens** [comment, marked as answer]
> My guess is that this isn't configurable because of time constrains, if that's the case, maybe I can give you a hand with that?

This was intentional given that `308` is the correct status code to use nowadays, see the documentation: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/308. 301 has some flaws like form data not going through.

**Comment by bericp1** [comment]
Even though 308 is definitely the appropriate status code to use technically, unfortunately for us at least, Google doesn't seem to index 308 as permanent despite various online threads on twitter and in blog posts. We turned on `trailingSlashes` a couple of weeks ago, and we can see that we're correctly serving 308s to google, but Google is indexing them as temporary redirects at least according to search console.

![](https://d.pr/i/vdBsjb.png)

![](https://d.pr/i/LsUP7I.png)

We're going to workaround this via nginx most likely to force these to be 301s instead and see how search console reacts. Search console _may_ be wrong because it doesn't have a way to show 308s in the console maybe but in that case I'd expect it instead to show them as permanent 301s if it were treating them as such?

I realize this isn't a nextjs bug hence no new issue but it would be great to be able to configure this redirect until such time that google publicly and directly supports 308.

Thanks!

**Reply by carl0s** [reply]
What happens when I use vercel to host Nextjs apps? Search console is alert me that there are a lot of indexes urls with 308 rather than 301.

**Comment by adrienharnay** [comment]
Hello, adding my input on this one. I also think it should be great to be able to configure the status code when redirecting (301 or 308, 302 or 307), because I'm not 100% sure the two work exactly the same SEO-wise (not sure they transfer the same SEO juice/weight, backlinks, etc).

We're using only GET requests, and our SEO agency has asked us several times to go back to 301 redirects. Would you consider adding an option for this? Thanks in advance!

**Comment by kyle-ssg** [comment]
We have a very high Google ranking for a few pages and this 308 redirect is hurting us a bit having migrated to NextJS+Vercel because it's indexing trailing / as duplicate pages. Is there a solution for this ? I'm going to look at patching wherever this is happening, presumably in the next node module somewhere.

**Reply by YazanAliDP** [reply]
any updates, I'm going through the same issue

**Comment by 3v1lm0nk3y** [comment]
Any updates here?
